src/Build/Evaluation/IntrinsicFunctions.cs

```diff
@@ -632,7 +632,9 @@ internal static bool AreFeaturesEnabled(Version wave)
         internal static string SubstringByTextElements(string input, int start, int length)
         {
             StringInfo stringInfo = new StringInfo(input);
-            return stringInfo.SubstringByTextElements(start, length);
+            if (stringInfo.LengthInTextElements > length + start)
+                return stringInfo.SubstringByTextElements(start, length);
```

Comment:
nit: we use braces for single line conditionals as well

```suggestion
            if (stringInfo.LengthInTextElements > length + start)
            {
                return stringInfo.SubstringByTextElements(start, length);
            }
```